src/components/fields/ArrayField.js

```diff
@@ -105,6 +105,16 @@ class ArrayField extends Component {
     };
   };
 
+  onReorderClick = (index, newIndex) => {
+    return (event) => {
+      event.preventDefault();
+      const items = this.state.items.slice(0);
+      const item = items.splice(index, 1)[0];
+      items.splice(newIndex, 0, item);
```

Comment:
We may want to move this as a helper in the `utils` module. Also you may want to factor the code [here](https://github.com/mozilla-services/react-jsonschema-form/blob/15c3f4b81edfd01a5241750166ba3d15d57f99ed/src/components/widgets/CheckboxesWidget.js#L6-L9) in that helper and use it in this implementation too.